### ISSUE TITLE
Console tests don't verify actual In/Out

### ISSUE THREAD

**Issue by ianhays** [dotnet/corefx#9777]
The tests in System.Console for the TextReader and TextWriter returned from Console.In and Console.Out don't verify anything about the non-redirected streams. Each test first calls `SetIn` or `SetOut` on a StreamWriter/Reader wrapped around a MemoryStream. This leads to issues like #9767 where basic operations have bugs that are supposedly tested against.

For example, we have this test:
```
[Fact]
public void Peek()
{
    const string expected = "ABC";
    Test(expected, () => {
        foreach (char expectedChar in expected)
        {
            Assert.Equal(expectedChar, Console.In.Peek());
            Assert.Equal(expectedChar, Console.In.Read());
        }
    });
}
```
that *seems* to be testing exactly what #9767 proved broken. However, in the `Test` helper we call `Console.SetIn(sr)` - where `sr` is a StreamReader around a MemoryStream - before performing the test action.

So this test is actually only verifying that `Console.In` roundtrips with `Console.SetIn` and further that a StreamWriter's Peek and Read functions work. 

TL;DR: We should look for a better way to test the Writer/Reader returned from a non-redirected Console.Out/Console.In.
